src/models/event-timeline-set.js

```diff
@@ -166,21 +167,35 @@ EventTimelineSet.prototype.resetLiveTimeline = function(backPaginationToken, flu
         newTimeline = this.addTimeline();
     }
 
-    // initialise the state in the new timeline from our last known state
-    const evMap = this._liveTimeline.getState(EventTimeline.FORWARDS).events;
-    const events = [];
-    for (const evtype in evMap) {
-        if (!evMap.hasOwnProperty(evtype)) {
-            continue;
-        }
-        for (const stateKey in evMap[evtype]) {
-            if (!evMap[evtype].hasOwnProperty(stateKey)) {
+    // move the state objects over from the old live timeline, then we'll
+    // keep using the same RoomMember objects for the 'live' set of members
+    newTimeline._startState = this._liveTimeline._startState;
+    newTimeline._endState = this._liveTimeline._endState;
+
+    if (!resetAllTimelines) {
+        // Now regenerate the state for the previously-live timeline, because
+        // we just stole it and put it on the new live timeline
+        // (If we're resetting all timelines, don't bother because the old live
+        // timeline is about to be thrown away anyway.
+        this._liveTimeline._startState = new RoomState(this._roomId);
+        this._liveTimeline._endState = new RoomState(this._roomId);
+        const evMap = this._liveTimeline.getState(EventTimeline.FORWARDS).events;
```

Comment:
Aren't we doing this too late? It'll be an empty map since we just clobbered the state?